crates/pop-cli/src/commands/call/chain.rs

```diff
@@ -289,26 +289,20 @@ impl CallChainCommand {
 
 	// Checks if the chain has the Sudo pallet and prompts the user to confirm if they want to
 	// execute the call via `sudo`.
-	fn configure_sudo(&mut self, chain: &Chain, cli: &mut impl Cli) -> Result<()> {
+	fn check_sudo(&mut self, chain: &Chain, cli: &mut impl Cli) -> Result<()> {
 		match find_dispatchable_by_name(&chain.pallets, "Sudo", "sudo") {
-			Ok(_) =>
-				if !self.sudo {
-					self.sudo = cli
-						.confirm(
-							"Would you like to dispatch this function call with `Root` origin?",
-						)
-						.initial_value(false)
-						.interact()?;
-				},
+			Ok(_) => {
+				self.sudo = cli
+					.confirm(
+						"Are you sure you want to dispatch this function call with `Root` origin?",
+					)
+					.initial_value(false)
```

Comment:
Nitpick: When the user passes `--sudo`, the confirmation prompt should default to `Yes`. 

Also, if `self.skip_confirm` is true, we should not prompt at all. This is important for example for a CI, where `--skip-confirm` avoids blocking on interactive input.